### ISSUE TITLE
Support regular fine-tuning alongside dreambooth

### ISSUE THREAD

**Issue by Hellisotherpeople** [d8ahazard/sd_dreambooth_extension#741]
Right now, within the Automatic11111 webUI, by default a user can create and train hypernetworks or textual inversion embeddings. 
With this extension, users have access to DreamBooth, Lora, and Imagic, which are all unique ways of fine-tuning models. DreamBooth tries hard to do prior preservation, Lora tries hard to work with low VRAM systems, and Imagic tries to work with a single image. 

But, regular fine-tuning is missing from both this extension and the Automatic11111 webui. In fact, there appear to be ZERO GUI based ways to do this on Linux right now. That's a pretty massive oversight. 

**WHAT**
Add traditional fine-tuning support, ala what is currently supported by https://github.com/victorchall/EveryDream2trainer or by Stable Trainer.

**Comment by ExponentialML** [comment]
While I agree it would be a nice addition, it would take quite a bit if work and time to implement in this repository (Like the EveryDream2trainer).

You can effectively achieve similar results as full fine tuning (which is kind of what Dreambooth is without prior preservation loss) by using the repository as is by:

1. Creating a dataset with images and captions.
2. Training with `[filewords]` as the `instance prompt`.
3. *Zero* prior preservation loss. You can use it for fine tuning, but it would be very complex matching every single class instance that you're trying to train.
4. Setting your image epochs high with a low learning rate (5e-6 or refer to `yaml` files on the training repositories).
5. Setting the proper Learning Rate scheduler (not really needed for AdamW, which is what this repository uses). 

I *don't* know if the VAE is trained in this repository, but the ones that are out there (by StabilityAI for instance) that should suffice for the majority of use cases.

**Comment by d8ahazard** [comment]
Actually, regular fine tuning is just dream booth without prior
preservation and longer captions.

Unless I'm mistaken, in which case I'm happy to implement whatever we
consider actual fine tuning, but after looking at the code for several
other projects, I'm pretty sure that we're already doing this.

**Comment by ZeroCool940711** [comment]
@d8ahazard It seems like what makes `EveryDreamTrainer`, well, `EveryDreamTrainer` is the `DataLoader`, if you can copy that you pretty much have `EveryDreamTrainer` implemented with `Dreambooth+LORA` and according to hlky which implemented this for [Nataili](https://github.com/Sygil-Dev/nataili/tree/v0.2.0) the dataloader can be a drop-in replacement for some stuff on dreambooth. Another thing that makes everydream work is that they also allow you to load the tags/captions from a txt file with the same filename as the images you are using for training instead of just using the filename of the images, this allows you to add way more tags and captions to use for training than what you can do with just the filename as there is a limit on what you can put in the filename and the formatting of it.

**Comment by d8ahazard** [comment]
> @d8ahazard It seems like what makes `EveryDreamTrainer`, well, `EveryDreamTrainer` is the `DataLoader`, if you can copy that you pretty much have `EveryDreamTrainer` implemented with `Dreambooth+LORA` and according to hlky which implemented this for [Nataili](https://github.com/Sygil-Dev/nataili/tree/v0.2.0) the dataloader can be a drop-in replacement for some stuff on dreambooth. Another thing that makes everydream work is that they also allow you to load the tags/captions from a txt file with the same filename as the images you are using for training instead of just using the filename of the images, this allows you to add way more tags and captions to use for training than what you can do with just the filename as there is a limit on what you can put in the filename and the formatting of it.

I mean, you realize that with the current dataloader I have implemented in Dreambooth; we can already do this, right?

I support image bucketing - using images of varying resolutions during training by grouping them into "buckets" of the same size.

We have supported reading captions from files for a while now. If a txt file exists with the same name as the instance image, it will read the txt file and use the contents as the prompt.

Additionally, we support tag shuffling - splitting tags by commas and randomizing the order...padding, cropping, and randomly horizontally flipping the images before training...

**Comment by ZeroCool940711** [comment]
Then I guess that would make your extension have the same as EveryDreamTrainer, pretty such that is all is needed for it to be the same. I might be wrong tho.

**Comment by ExponentialML** [comment]
Completely understand your sentiments @Hellisotherpeople .

> I'm sure that prior preservation techniques used by Dreambooth have at least some kind of downsides compared to turning it off...

Prior preservation is actually really cool, just kind of complex to setup if you're doing a full fine tune (each image would need a class, which would mostly be guesstimates at best depending on the model).
 
It's mainly to prevent language drift and your trained data from leaking into other classes (trained dog without prior loss = cats and people may look like your trained dog). 

I would consider this a downside if you were trying to train the entire model to emulate a certain look, style, objects, etc.
